custom_model_runner/datarobot_drum/drum/adapters/model_adapters/python_model_adapter.py

```diff
@@ -107,43 +104,48 @@ def __init__(self, model_dir, target_type=None):
         # New custom task class and instance loaded from custom.py
         self._custom_task_class = None
         self._custom_task_class_instance = None
-        self._guard_moderation_hooks = None
-        self._guard_pipeline = None
+        self._moderation_pipeline = None
+        self._moderation_score_hook = None
+        self._moderation_chat_hook = None
 
         if target_type in (TargetType.TEXT_GENERATION, TargetType.VECTOR_DATABASE):
             self._target_name = os.environ.get("TARGET_NAME")
             if not self._target_name:
                 raise ValueError(
                     "Unexpected empty target name for text generation or vector database target."
                 )
-            if target_type == TargetType.TEXT_GENERATION:
-                self._load_guard_hooks_for_drum()
+            self._load_moderation_hooks()
         else:
             self._target_name = None
 
-    def _load_guard_hooks_for_drum(self):
+    def _load_moderation_hooks(self):
         try:
-            guard_module = __import__(GUARD_HOOK_MODULE, fromlist=[MODERATIONS_LIBRARY_PACKAGE])
+            mod_module = __import__(MODERATIONS_HOOK_MODULE, fromlist=[MODERATIONS_LIBRARY_PACKAGE])
```

Comment:
We should start moving to OOP to have additional functionality to be imported as classes.
I suggest start adding(or even completely moving to OOP) such support into moderations.

Probably moderations would be a good candidate for POC as currently it is still relatively small.